fastlane/spec/plugins_specs/plugin_generator_spec.rb

```diff
@@ -122,7 +124,10 @@
         $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 
         # rubocop:disable Security/Eval
-        eval(plugin_rb_contents)
+        # starting with ruby 3.3 we must pass the __FILE__ of the actual file location for the all_class method implementation to work
+        # also we expand the relative_path within the chdir, as Dir.chdir on Mac will make it so tmp paths will always be under /private,
+        # while expand_path called from outside of the chdir will not prefix by /private
+        eval(plugin_rb_contents, binding, File.expand_path(relative_path), __LINE__)
```

Comment:
`__FILE__` or `__LINE__`? 👀 

Also some nitpicks on wording:

```suggestion
        # Starting with Ruby 3.3, we must pass the __FILE__ of the actual file location for the all_class method implementation to work.
        # Also, we expand the relative_path within Dir.chdir, as Dir.chdir on macOS will make it so tmp paths will always be under /private,
        # while expand_path called from outside of Dir.chdir will not be prefixed by /private
        eval(plugin_rb_contents, binding, File.expand_path(relative_path), __LINE__)
```

Reply:
> __FILE__ or __LINE__? 👀

We pass `File.expand_path(relative_path)` so that the `__FILE__` value within the [evaluated script](https://github.com/fastlane/fastlane/blob/9c4a7b4dbfecd0982a7fd26820ab72784c56dfdc/fastlane/lib/fastlane/plugins/template/lib/fastlane/plugin/%25plugin_name%25.rb.erb#L7) has the correct value. This is really specific for the tests, so that they pass.

See
```C
return eval_string(self, src, scope, file, line); 
```
in the source implementation of [`eval`](https://apidock.com/ruby/v2_5_5/Kernel/eval)